### PR TITLE
Implement issue #20

### PR DESCRIPTION
**Pull request by Trifall**
This PR addresses the issue `Shortcut to change chat size dynamically` #20 

I have added a ResizeAuxiliaryBarWidthAction in `auxiliaryBarActions.ts` for the chat bar (auxiliary bar) to expand it's size when pressed, and then revert to the smaller original size when pressed. The left side-bar is hidden when the chat bar is expanded, and is returned to its previous state upon the chat bar being reverted to its original size.

Primary Shortcut: The primary keyboard shortcut is `Ctrl (or Cmd on macOS) + [`.
Linux: On Linux, the primary shortcut is `CTRL + [`
macOS: On macOS, the primary shortcut is `CTRL + [`.
Windows: On Windows, the primary shortcut is `ALT + [`.

The max width of the chat bar is fixed at 70% the width of the App's inner window, the minimum width is 600px; the defaulted value is `2 * currentChatBarWidth`. The maximum width rule takes precedence over the minimum width rule. These sizes are arbitrary, but they look solid imo. Those preset values can be changed in the minWidth and maxWidth variables respectively.

Demo:
https://github.com/trypear/pearai-app/assets/13270623/937b0bba-bdc7-481b-b8d9-4a103ac480d5

### CODE REVIEW

**Comment by cwclancy** [comment]
Can you add pics as a test plan please :)

**Review by cwclancy — COMMENTED** [review]
like comment on main, please add picture/video to test plan so easier to tell what's happening, but code looks reasonable

**Comment by Trifall** [comment]
Added video demo of the functionality to original comment

**Review by cwclancy — APPROVED** [review]
nice, this is sweet!

File: src/vs/workbench/browser/parts/auxiliarybar/auxiliaryBarActions.ts [hunk 1]
```diff
@@ -145,3 +145,85 @@ MenuRegistry.appendMenuItems([
...
+
+		// If the current width is less than or equal to the previous width, expand the auxiliary bar
+		if (auxBarDimensions.width <= ResizeAuxiliaryBarWidthAction._previousAuxiliaryBarWidth) {
+			// Expand to the calculated minWidth
+			layoutService.resizePart(Parts.AUXILIARYBAR_PART, (minWidth - auxBarDimensions.width), 0);
+			// Hide the left side bar if it was previously visible
+			layoutService.setPartHidden(true, Parts.SIDEBAR_PART);
+		} else {
+			// If the current width is greater than the previous width, collapse the auxiliary bar back to the previous width (initial width)
+			layoutService.resizePart(Parts.AUXILIARYBAR_PART, (auxBarDimensions.width - ResizeAuxiliaryBarWidthAction._previousAuxiliaryBarWidth) * -1, 0);
```
Comment (Fryingpannn) [inline src/vs/workbench/browser/parts/auxiliarybar/auxiliaryBarActions.ts:218]: Wonder if `(auxBarDimensions.width - ResizeAuxiliaryBarWidthAction._previousAuxiliaryBarWidth` can directly just be set to `ResizeAuxiliaryBarWidthAction._previousAuxiliaryBarWidth` when resizing?

**Review by Fryingpannn — APPROVED** [review]
Beautiful solution. This is a first step in differentiating from competitors. 🎉

File: src/vs/workbench/browser/parts/auxiliarybar/auxiliaryBarActions.ts (hunk 1 above)
Comment (Trifall) [reply src/vs/workbench/browser/parts/auxiliarybar/auxiliaryBarActions.ts:218]: It's possible. When I looked around I didn't immediately see a 'set to fixed value' method for the part width, and resizePart takes width and height as parameters, but its +/-, so it just adds or subtracts the value from the current width/height.

**Pull request merged**